graphql/util.go

```diff
@@ -58,40 +58,77 @@ func getGroupedFiles(ctx context.Context, name string, taskID string, execution
+func setManyTasksScheduled(ctx context.Context, url string, isActive bool, taskIDs ...string) ([]*restModel.APITask, error) {
 	usr := mustHaveUser(ctx)
-	t, err := task.FindOneId(taskID)
+	tasks, err := task.FindAll(db.Query(task.ByIds(taskIDs)))
 	if err != nil {
 		return nil, ResourceNotFound.Send(ctx, err.Error())
 	}
-	if t == nil {
-		return nil, ResourceNotFound.Send(ctx, errors.Errorf("task %s not found", taskID).Error())
+	if len(tasks) != len(taskIDs) {
+		foundTaskIds := []string{}
+		for _, ft := range tasks {
+			foundTaskIds = append(foundTaskIds, ft.Id)
+		}
+		missingTaskIds := sliceDifference(taskIDs, foundTaskIds)
+		grip.Error(message.Fields{
+			"message":      "could not find all tasks",
+			"missingTasks": missingTaskIds,
+		})
+		return nil, ResourceNotFound.Send(ctx, errors.New("tasks not found").Error())
 	}
-	if t.Requester == evergreen.MergeTestRequester && isActive {
-		return nil, InputValidationError.Send(ctx, "commit queue tasks cannot be manually scheduled")
+	for _, t := range tasks {
+		if t.Requester == evergreen.MergeTestRequester && isActive {
+			return nil, InputValidationError.Send(ctx, "commit queue tasks cannot be manually scheduled")
+		}
 	}
-	if err = model.SetActiveState(t, usr.Username(), isActive); err != nil {
+	if err = model.SetActiveState(usr.Username(), isActive, tasks...); err != nil {
 		return nil, InternalServerError.Send(ctx, err.Error())
 	}
 
-	// Get the modified task back out of the db
-	t, err = task.FindOneId(taskID)
+	// Get the modified tasks back out of the db
+	tasks, err = task.FindAll(db.Query(task.ByIds(taskIDs)))
 	if err != nil {
 		return nil, ResourceNotFound.Send(ctx, err.Error())
 	}
-	if t == nil {
-		return nil, ResourceNotFound.Send(ctx, err.Error())
+	if len(tasks) != len(taskIDs) {
+		foundTaskIds := []string{}
+		for _, ft := range tasks {
+			foundTaskIds = append(foundTaskIds, ft.Id)
+		}
+		missingTaskIds := sliceDifference(taskIDs, foundTaskIds)
+		grip.Error(message.Fields{
+			"message":        "could not find all tasks",
+			"missingTaskIds": missingTaskIds,
+		})
+		return nil, ResourceNotFound.Send(ctx, errors.New("tasks not found").Error())
 	}
-	apiTask := restModel.APITask{}
-	err = apiTask.BuildFromService(t)
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, err.Error())
+	apiTasks := []*restModel.APITask{}
+	for _, t := range tasks {
+		apiTask := restModel.APITask{}
+		err = apiTask.BuildFromService(&t)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		err = apiTask.BuildFromService(url)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		apiTasks = append(apiTasks, &apiTask)
 	}
-	err = apiTask.BuildFromService(url)
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, err.Error())
+	return apiTasks, nil
+}
+
+func sliceDifference(a, b []string) (diff []string) {
```

Comment:
We actually already have utility.StringSliceSymmetricDifference

Reply:
Was looking for something like this but didn't find it. Thanks!